### PR TITLE
Add OneLake datastore

### PR DESCRIPTION
**Pull request by youngpark**
# Description

This pull request adds support for OneLake as a datastore type and also adds tests to validate the new changes.

# All SDK Contribution checklist:
- [X] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [X] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_datastore/one_lake.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    )
+    # required fields for OneLake
+    one_lake_workspace_name = fields.Str(required=True)
+    artifact = NestedField(OneLakeArtifactSchema)
+    endpoint = fields.Str()
+    # ServicePrincipal and UserIdentity are the two supported credential types
+    credentials = UnionField(
+        [
+            NestedField(ServicePrincipalSchema),
+            NestedField(NoneCredentialsSchema),
```
Comment (diondrapeck) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_datastore/one_lake.py:38]: Is NoneCredentialsSchema the same as UserIdentity?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_datastore/one_lake.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+from azure.ai.ml.entities._util import load_from_dict
+
+
+class OneLakeDatastore(Datastore):
+    """OneLake datastore that is linked to an Azure ML workspace.
+
+    :param name: Name of the datastore.
+    :type name: str
+    :param artifact: OneLake Artifact. Only LakeHouse artifacts are currently supported.
+    :type artifact: OneLakeArtifact
```
Comment (diondrapeck) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_datastore/one_lake.py:25]: We're working to improve documentation standards across the SDK. One way we're doing that is by making sure documentation is linked together. Can you replace the types here with their full paths in this format: 

```
OneLakeArtifact -> ~azure.ai.ml.entities.OneLakeArtifact
ServicePrincipalConfiguration -> ~azure.ai.ml.entities.ServicePrincipalConfiguration
NoneCredentialConfiguration -> ~azure.ai.ml.entities.NoneCredentialConfiguration
```

Comment (youngpark) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_datastore/one_lake.py]: Thanks - updated!

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_datastore/one_lake.py (hunk 1 above)
Comment (youngpark) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_datastore/one_lake.py:38]: Yup, my understanding is that there isn't one for UserIdentity and NoneCredentialsSchema is used to indicate using UserIdentity for the supported cases.

**Review by diondrapeck — APPROVED** [review]

**Pull request merged**